### PR TITLE
fix: upgrade proxy-agent to ^6.3.0 to remove vm2 deps

### PR DESCRIPTION
**Pull request by awsluja**
#### Description of changes
E2e run https://tiny.amazon.com/1gmqt246c/IsenLink

This removes the last of the vm2 dependencies by upgrading proxy-agent.
Proxy agent constructor has changed since v5, so there were updates as well.
See proxy-agent docs here: https://github.com/TooTallNate/proxy-agents/tree/main/packages/proxy-agent

Proxy-agent uses this module to determine which proxy to use, so we don't need to manually specify http/https anymore, see:
https://www.npmjs.com/package/proxy-from-env
https://www.npmjs.com/package/proxy-from-env#environment-variables

We do still need to conditionally check if proxy vars are set to decide if we want to initialize a new proxyagent.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -31455,18 +31474,6 @@ node-pty@beta:
   languageName: node
   linkType: hard
 
-"vm2@npm:^3.9.3":
```
Comment (awsluja) [inline yarn.lock:31458]: vm2 removal is here

File: packages/amplify-category-notifications/src/pinpoint-helper.ts [hunk 2]
```diff
@@ -547,7 +547,7 @@ export const getPinpointClient = async (
   if (httpProxy) {
     aws.config.update({
       httpOptions: {
-        agent: proxyAgent(httpProxy),
+        agent: new ProxyAgent(),
```
Comment (sobolk) [inline packages/amplify-category-notifications/src/pinpoint-helper.ts:550]: is it ok to drop `httpProxy` param here ?

it seems to be a pattern in the PR.

**Review by sobolk — APPROVED** [review]

File: packages/amplify-category-notifications/src/pinpoint-helper.ts (hunk 2 above)
Comment (awsluja) [reply packages/amplify-category-notifications/src/pinpoint-helper.ts:550]: yeah see description for more details; tldr it reads from process.env by default so we don't need to manually specify anymore

Comment (sobolk) [reply packages/amplify-category-notifications/src/pinpoint-helper.ts:550]: if we read these variables anyway for the `if` I'd rather pass this explicitly.
unless they took away API to do this. In that case might be worth to add comment for future reader why `httpProxy` is not consumed.

Comment (awsluja) [reply packages/amplify-category-notifications/src/pinpoint-helper.ts:551]: added notes

**Review by sobolk — APPROVED** [review]

File: shared-scripts.sh [hunk 3]
```diff
@@ -125,6 +125,7 @@ function _verifyAPIExtract {
     # download [repo, .cache from s3]
     loadCache repo $CODEBUILD_SRC_DIR
     loadCache .cache $HOME/.cache
+    unset IS_AMPLIFY_CI
```
Comment (jhockett) [inline shared-scripts.sh:128]: Why is this included in this PR?

Comment (awsluja) [reply shared-scripts.sh:128]: short story: codebuild sets IS_AMPLIFY_CI globally, but circleci doesn't, so this achieves parity.
long story:
this PR changed the API slightly, which shined a light on something we didn't see before in codebuild, but happens all the time in circleci. That is, verify-api-extract actually downloads the amplify-cli & installs it due to a post install script for the amplify-cli-npm package. Downloading the CLI has no effect on the API step, but when because its running in codebuild CI - the IS_AMPLIFY_CI flag makes it try to download from s3 (pkg cli) using the binary hash. CircleCI does not set IS_AMPLIFY_CI for the verify-api-extract, so it succeeds there because it just downloads the production pkg cli. Download the pkg cli has no affect on verify-api-extract's result.

Comment (jhockett) [reply shared-scripts.sh:128]: Would you mind doing a quick one-off PR and merging it first? It will improve discoverability for the change and then this comment can become the PR description.

Comment (awsluja) [reply shared-scripts.sh:128]: https://github.com/aws-amplify/amplify-cli/pull/13019

**Review by jhockett — APPROVED** [review]
